pkg/build/api/types.go

```diff
@@ -911,6 +915,9 @@ type GenericWebHookEvent struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar
+
+	// StrategyOptions contains additional strategy-specific options for the build
+	StrategyOptions *BuildStrategyOptions
```

Comment:
DockerStrategy doesn't sound like a good name for a field that doesn't hold a DockerStrategy object.  That is what i'd expect a field with that name to hold.